src/Microsoft.Sbom.Api/Executors/PackageInfoJsonWriter.cs

```diff
@@ -66,7 +66,15 @@ private async Task GenerateJson(
             {
                 var generationResult =
                     manifestGeneratorProvider.Get(sbomConfig.ManifestInfo).GenerateJsonDocument(packageInfo);
-                sbomConfig.Recorder.RecordPackageId(generationResult?.ResultMetadata?.EntityId, generationResult?.ResultMetadata?.DependOn);
+
+                if (generationResult?.ResultMetadata?.DependOn != null)
+                {
+                    foreach (var dependency in generationResult?.ResultMetadata?.DependOn)
+                    {
+                        sbomConfig.Recorder.RecordPackageId(generationResult?.ResultMetadata?.EntityId, dependency);
+                    }
+                }
```

Comment:
This is the only change to the API method. Since the method header itself has not changed, I think this is a non-breaking API change but would appreciate a double confirmation.

Reply:
It's technically still a breaking change, since it's changing a type in the data contract. That said, we don't really expect people to interact with this type directly, so I think we can treat it as non-breaking.